src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java

```diff
@@ -531,12 +532,13 @@ private TaskAbortCause getTaskAbortCause(Task task) {
 
     // region purge locks
     @Override
-    public boolean purgeTask(String chainTaskId) {
+    public boolean purgeTask(final String chainTaskId) {
         taskAccessForNewReplicateLocks.remove(chainTaskId);
         return !taskAccessForNewReplicateLocks.containsKey(chainTaskId);
     }
 
     @Override
+    @PreDestroy
     public void purgeAllTasksData() {
         taskAccessForNewReplicateLocks.clear();
```

Comment:
```suggestion
        log.info("Method purgeAllTasksData() called to perform task data cleanup.");
        taskAccessForNewReplicateLocks.clear();
```

Reply:
resolved in : [a35058a](https://github.com/iExecBlockchainComputing/iexec-core/pull/724/commits/a35058a652778f31cfebf678aab460f24ca556b8)